File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        res = 0
        for i, num in enumerate(nums_set):
            if num - 1 not in nums_set:
                tempres = 0
                temp = num
                while temp in nums_set:
                    tempres += 1
                    temp += 1
                if tempres > res:
                    res = tempres
        return res

File: Data Structures & Algorithms/top-k-elements-in-list/submission-6.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        Dictionary<int, int> counter = new Dictionary<int, int>();
        foreach (int num in nums) {
            if (!counter.ContainsKey(num)) {
                counter[num] = 0;
            }
            counter[num]++;
        }
        List<KeyValuePair<int, int>> freq_list = counter.ToList();
        freq_list.Sort((a,b) => b.Value.CompareTo(a.Value));
        int[] res = new int[k];
        for (int i = 0; i < k; i++) {
            res[i] = freq_list[i].Key;
        }
        return res;
    }
}
